Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.cpp

```diff
@@ -6,208 +6,105 @@
+            if (m_verbose)
             {
-                auto registryIter = registry->m_registries.find(name.c_str());
-                if (registryIter != registry->m_registries.end())
-                {
-                    registryIter->second->Init(nodeRegistry);
-                }
+                AZ_Info("ScriptCanvas", "Register Descriptor: %s", descriptor->GetName());
             }
-        }
-    }
 
-    AZStd::vector<AZ::ComponentDescriptor*> AutoGenRegistryManager::GetComponentDescriptors()
-    {
-        AZStd::vector<AZ::ComponentDescriptor*> descriptors;
-        if (auto registry = GetInstance())
-        {
-            for (auto& iter : registry->m_registries)
-            {
-                if (iter.second)
-                {
-                    auto nodeableDescriptors = iter.second->GetComponentDescriptors();
-                    descriptors.insert(descriptors.end(), nodeableDescriptors.begin(), nodeableDescriptors.end());
-                }
-            }
         }
-        return descriptors;
     }
+}
 
-    AZStd::vector<AZ::ComponentDescriptor*> AutoGenRegistryManager::GetComponentDescriptors(const char* registryName)
+bool ScriptCanvasModel::RegisterReflection(const AZStd::string& name, ReflectFunction reflect, AZ::ComponentDescriptor* descriptor/* = nullptr*/)
+{
+    if (!m_registeredReflections.contains(name))
     {
-        AZStd::vector<AZ::ComponentDescriptor*> descriptors;
-        if (auto registry = GetInstance())
+        if (descriptor)
         {
-            auto registryNames = registry->GetRegistryNames(registryName);
-            for (auto name : registryNames)
-            {
-                auto registryIter = registry->m_registries.find(name.c_str());
-                if (registryIter != registry->m_registries.end())
-                {
-                    auto registryDescriptors = registryIter->second->GetComponentDescriptors();
-                    descriptors.insert(descriptors.end(), registryDescriptors.begin(), registryDescriptors.end());
-                }
-            }
-        }
-        return descriptors;
-    }
+            m_descriptors.push_back(descriptor);
 
-    void AutoGenRegistryManager::Reflect(AZ::ReflectContext* context)
-    {
-        if (auto registry = GetInstance())
-        {
-            for (auto& iter : registry->m_registries)
+            if (m_verbose)
             {
-                if (iter.second)
-                {
-                    iter.second->Reflect(context);
-                }
+                AZ_Info("ScriptCanvas", "RegisterReflection Descriptor: %s", name.c_str());
             }
         }
-    }
-
-    void AutoGenRegistryManager::Reflect(AZ::ReflectContext* context, const char* registryName)
-    {
-        if (auto registry = GetInstance())
+        else
         {
-            auto registryNames = registry->GetRegistryNames(registryName);
-            for (auto name : registryNames)
+            m_registeredReflections[name] = reflect;
```

Comment:
So the reflect function isn't used if a Descriptor is supplied.

This should be added as a doxygen comment in the header for this RegisterReflection function.

Future maintainers could use this information spelled out for them. that this function can register either a component descriptor or a functor which accepts a ReflectContext pointer.